junit-jupiter-api/src/main/java/org/junit/jupiter/api/extension/InvocationInterceptor.java

```diff
@@ -156,11 +156,29 @@ default void interceptTestTemplateMethod(Invocation<Void> invocation,
 	 * {@code null}
 	 * @param extensionContext the current extension context; never {@code null}
 	 * @throws Throwable in case of failures
+	 * @deprecated use {@link #interceptDynamicTest(Invocation, DynamicTestInvocationContext, ExtensionContext)} instead
 	 */
+	@Deprecated
 	default void interceptDynamicTest(Invocation<Void> invocation, ExtensionContext extensionContext) throws Throwable {
 		invocation.proceed();
 	}
 
+	/**
+	 * Intercept the invocation of a {@link DynamicTest}.
+	 *
+	 * @param invocation the invocation that is being intercepted; never
+	 * {@code null}
+	 * @param invocationContext the context of the invocation that is being
+	 * intercepted; never {@code null}
+	 * @param extensionContext the current extension context; never {@code null}
+	 * @throws Throwable in case of failures
+	 */
+	default void interceptDynamicTest(Invocation<Void> invocation, DynamicTestInvocationContext invocationContext,
```

Comment:
```suggestion
	@API(status = EXPERIMENTAL, since = "5.8")
	default void interceptDynamicTest(Invocation<Void> invocation, DynamicTestInvocationContext invocationContext,
```